pkg/actors/targets/workflow/orchestrator/run.go

```diff
@@ -132,122 +127,56 @@ func (w *workflow) runWorkflow(ctx context.Context, reminder *actorapi.Reminder)
+		if err = o.createTimers(ctx, rs.GetPendingTimers(), state.Generation); err != nil {
+			executionStatus = diag.StatusRecoverable
+			return todo.RunCompletedFalse, wferrors.NewRecoverable(err)
 		}
 	}
 
-	err = w.callActivities(ctx, rs.GetPendingTasks(), state.Generation)
+	err = o.callActivities(ctx, rs.GetPendingTasks(), state.Generation)
 	if err != nil {
 		executionStatus = diag.StatusRecoverable
 		return todo.RunCompletedFalse, err
 	}
 
-	var (
-		compl todo.RunCompleted
-		wg    sync.WaitGroup
-		lock  sync.Mutex
-		errs  []error
-	)
-
-	for method, msgList := range reqsByName {
-		wg.Add(len(msgList))
-		for _, msg := range msgList {
-			go func(method string, msg *backend.OrchestrationRuntimeStateMessage) {
-				defer wg.Done()
-
-				var requestBytes []byte
-				var perr error
-				if method == todo.CreateWorkflowInstanceMethod {
-					requestBytes, perr = proto.Marshal(&backend.CreateWorkflowInstanceRequest{
-						StartEvent: msg.GetHistoryEvent(),
-					})
-					if perr != nil {
-						lock.Lock()
-						errs = append(errs, fmt.Errorf("failed to marshal createWorkflowInstanceRequest: %w", perr))
-						compl = todo.RunCompletedTrue
-						lock.Unlock()
-						return
-					}
-				} else {
-					requestBytes, perr = proto.Marshal(msg.GetHistoryEvent())
-					if perr != nil {
-						lock.Lock()
-						errs = append(errs, perr)
-						compl = todo.RunCompletedTrue
-						lock.Unlock()
-						return
-					}
-				}
+	// Process the outbound orchestrator events
+	var addWorkflows []*backend.OrchestrationRuntimeStateMessage
+	var createWorkflows []*backend.OrchestrationRuntimeStateMessage
+	for _, msg := range rs.GetPendingMessages() {
+		switch {
+		case msg.GetHistoryEvent().GetExecutionStarted() != nil:
+			createWorkflows = append(createWorkflows, msg)
 
-				log.Debugf("Workflow actor '%s': invoking method '%s' on workflow actor '%s'", w.actorID, method, msg.GetTargetInstanceID())
+		case msg.GetHistoryEvent().GetSubOrchestrationInstanceCompleted() != nil, msg.GetHistoryEvent().GetSubOrchestrationInstanceFailed() != nil:
+			addWorkflows = append(addWorkflows, msg)
 
-				_, eerr := w.router.Call(ctx, internalsv1pb.
-					NewInternalInvokeRequest(method).
-					WithActor(w.actorType, msg.GetTargetInstanceID()).
-					WithData(requestBytes).
-					WithContentType(invokev1.ProtobufContentType),
-				)
-				if eerr != nil {
-					lock.Lock()
-					executionStatus = diag.StatusRecoverable
-					errs = append(errs, fmt.Errorf("failed to invoke method '%s' on actor '%s': %w", method, msg.GetTargetInstanceID(), eerr))
-					lock.Unlock()
-					return
-				}
-			}(method, msg)
+		default:
+			return todo.RunCompletedTrue, fmt.Errorf("workflow actor '%s': don't know how to process outbound message '%v'", o.actorID, msg)
 		}
 	}
 
-	wg.Wait()
-	if len(errs) > 0 {
-		return compl, errors.Join(errs...)
+	if err = o.callAddEventStateMessage(ctx, addWorkflows); err != nil {
+		return todo.RunCompletedTrue, err
```

Comment:
Shouldn't this return false? I think before it returned false
```suggestion
		return todo.RunCompletedFalse, err
```